mmengine/config/new_config.py

```diff
@@ -0,0 +1,459 @@
+            # copied from setup.cfg
+            yapf_style = dict(
+                based_on_style='pep8',
+                blank_line_before_nested_class_or_def=True,
+                split_before_expression_after_opening_paren=True)
+            try:
+                text, _ = FormatCode(
+                    text, style_config=yapf_style, verify=True)
+            except:  # noqa: E722
+                raise SyntaxError('Failed to format the config file, please '
+                                  f'check the syntax of: \n{text}')
+
+        return text
+
+    def __getstate__(
+        self
+    ) -> Tuple[dict, Optional[str], Optional[str], bool]:  # type: ignore
+        return (self._cfg_dict, self._filename, self._text,
+                self._format_python_code)
+
+    def __setstate__(  # type: ignore
+        self,
+        state: Tuple[dict, Optional[str], Optional[str], bool],
+    ):
+        super(Config, self).__setattr__('_cfg_dict', state[0])
+        super(Config, self).__setattr__('_filename', state[1])
+        super(Config, self).__setattr__('_text', state[2])
+        super(Config, self).__setattr__('_format_python_code', state[3])
+
+    def _to_lazy_dict(self, keep_imported: bool = False) -> dict:
+        """Convert config object to dictionary and filter the imported
+        object."""
+        res = self._cfg_dict._to_lazy_dict()
+
+        if keep_imported:
+            return res
+        else:
+            return dict(filter(filter_imports, res.items()))
+
+    def to_dict(self, keep_imported: bool = False):
+        """Convert all data in the config to a builtin ``dict``.
+
+        Args:
+            keep_imported (bool): Whether to keep the imported field.
+                Defaults to False
+
+        If you import third-party objects in the config file, all imported
+        objects will be converted to a string like ``torch.optim.SGD``
+        """
+        _cfg_dict = self._to_lazy_dict(keep_imported=keep_imported)
+
+        def lazy2string(cfg_dict):
+            if isinstance(cfg_dict, dict):
+                return type(cfg_dict)(
+                    {k: lazy2string(v)
+                     for k, v in cfg_dict.items()})
+            elif isinstance(cfg_dict, (tuple, list)):
+                return type(cfg_dict)(lazy2string(v) for v in cfg_dict)
+            else:
+                dump_str = dump_extra_type(cfg_dict)
+                return dump_str if dump_str is not None else cfg_dict
+
+        return lazy2string(_cfg_dict)
+
+
+class BaseImportContext():
+
+    def __enter__(self):
+        # Disable enabled lazy loader during parsing base
+        self.lazy_importers = []
+        for p in sys.meta_path:
+            if isinstance(p, LazyImportContext) and p.enable:
+                self.lazy_importers.append(p)
+                p.enable = False
+
+        old_import = builtins.__import__
+
+        def new_import(name, globals=None, locals=None, fromlist=(), level=0):
```

Comment:
Suggest describing why do we need to replace `old_import` with `new_import` here. The description should be high-level to tell which syntax (relative import) is enhanced

Reply:
It seems that `new_import` means the enhanced `non_lazy_import` which supports  more `relative import` syntaxes. If I understand it correctly, suggest renaming it to `non_lazy_import`

Reply:
The context doesn't depend on lazy import. and the description is added.